x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java

```diff
@@ -125,25 +133,39 @@ public static void operation(PlanExecutor planExecutor, EqlSearchTask task, EqlS
         TimeValue timeout = TimeValue.timeValueSeconds(30);
         String clientId = null;
 
-        ParserParams params = new ParserParams(zoneId)
-            .fieldEventCategory(request.eventCategoryField())
-            .fieldTimestamp(request.timestampField())
-            .fieldTiebreaker(request.tiebreakerField())
-            .resultPosition("tail".equals(request.resultPosition()) ? Order.OrderDirection.DESC : Order.OrderDirection.ASC)
-            .size(request.size())
-            .fetchSize(request.fetchSize());
-
         RemoteClusterRegistry remoteClusterRegistry = new RemoteClusterRegistry(transportService.getRemoteClusterService(),
             request.indicesOptions());
-        EqlConfiguration cfg = new EqlConfiguration(request.indices(), zoneId, username, clusterName, filter,
+        Set<String> clusterAliases = remoteClusterRegistry.clusterAliases(request.indices(), false);
+        if (canMinimizeRountrips(request, clusterAliases)) {
+            String clusterAlias = clusterAliases.iterator().next();
+            String[] remoteIndices = new String[request.indices().length];
+            for (int i = 0; i < request.indices().length; i++) {
+                remoteIndices[i] = request.indices()[i].substring(clusterAlias.length() + /*`:`*/1);
```

Comment:
That inline comment is too l33t.  please move it after the statement or above it since as it stands requires multi-parsing.

Reply:
Pulled that into a trailing comment.